execute/table/copy.go

```diff
@@ -89,13 +101,16 @@ func (tb *tableBuffer) BufferN() int {
 }
 
 func (tb *tableBuffer) Copy() flux.BufferedTable {
-	for i := tb.i; i < len(tb.buffers); i++ {
+	if atomic.LoadInt32(&tb.used) == 1 {
+		panic("tried to copy an already used tableBuffer")
```

Comment:
I'm going to approve, but is there any chance you can have this panic only exist when the `assert` tag is used?

The `assert` tag is part of arrow for multiple releases and we use it when running tests.

Reply:
I'll look into that. You'd rather see some later downstream error or panic in these cases, assuming they reach production somehow?

Reply:
I switched this to be a logging call instead for now. Arrow's `debug.Assert()` is neat, but it lives in their `internal/` module space so I don't think we can access it.

We might want to add something like that to flux directly.